_query-dsl/specialized/percolate.md

```diff
@@ -0,0 +1,620 @@
+---
+layout: default
+title: Percolate
+parent: Specialized queries
+nav_order: 55
+---
+
+# Percolate
+
+Use the `percolate` query to find stored queries that match a given document. This is the opposite of a regular search. Instead of finding documents that match a query, you find queries that match a document. Percolate queries are often used for alerting, notifications, and reverse search use cases.
+
+- You can percolate a document provided inline or fetch an existing document from an index.
+- The document and the stored queries must use the same field names and types.
+- You can combine percolation with filtering and scoring to build complex matching systems.
+- Percolate queries are considered expensive and will only run if the cluster setting `search.allow_expensive_queries` is set to `true` (default). If this setting is `false`, percolate queries will be rejected.
+
+## How percolation works
+
+1. Saved queries are stored in a special [`percolator` field type]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/).
+2. Documents are compared against all saved queries.
+3. Matching queries are returned with their `_id`.
+4. If highlighting is enabled, matched text snippets are also returned.
+5. If multiple documents are sent, `_percolator_document_slot` displays which document matched.
+
+## Create an index to store saved queries
+
+First, create an index with `mappings` configured with [`percolator` field type]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/) to store the saved queries:
+
+```json
+PUT /my_percolator_index
+{
+  "mappings": {
+    "properties": {
+      "query": {
+        "type": "percolator"
+      },
+      "title": {
+        "type": "text"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add a query matching "apple" in `title` field:
+
+```json
+POST /my_percolator_index/_doc/1
+{
+  "query": {
+    "match": {
+      "title": "apple"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add a query matching "banana" in `title` field:
+
+```json
+POST /my_percolator_index/_doc/2
+{
+  "query": {
+    "match": {
+      "title": "banana"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Percolate an inline document
+
+You can test an inline document against saved queries:
+
+```json
+POST /my_percolator_index/_search
+{
+  "query": {
+    "percolate": {
+      "field": "query",
+      "document": {
+        "title": "Fresh Apple Harvest"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This will match the saved query looking for "apple":
```

Comment:
I think we should provide multi query match for the percolated documents as well to demonstrate the slot numbering